File: ArchitectureComponentsMVVM/app/src/main/java/com/andr/mvvm/RetrofitRoom/Repository/PostInfoRoomDataBase.java
package com.andr.mvvm.RetrofitRoom.Repository;

import android.arch.persistence.db.SupportSQLiteDatabase;
import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;
import android.support.annotation.NonNull;


import com.andr.mvvm.RetrofitRoom.Models.ResultModel;
import com.andr.mvvm.repository.UserDao;

@Database(entities = {ResultModel.class}, version = 1)
public abstract class PostInfoRoomDataBase extends RoomDatabase {
    public abstract PostInfoDao postInfoDao();

    private static PostInfoRoomDataBase INSTANCE;


    static PostInfoRoomDataBase getDatabase(final Context context) {
        if (INSTANCE == null) {
            synchronized (PostInfoRoomDataBase.class) {
                if (INSTANCE == null) {
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(),
                            PostInfoRoomDataBase.class, "postinfo_database")
                            .addCallback(sRoomDatabaseCallback)
                            .build();

                }
            }
        }
        return INSTANCE;
    }

    private static Callback sRoomDatabaseCallback =
            new Callback(){

                @Override
                public void onOpen (@NonNull SupportSQLiteDatabase db){
                    super.onOpen(db);
                    new PopulateDbAsync(INSTANCE).execute();
                }
            };
}

File: MVVM_Kotlin/app/src/main/java/com/mvvm/kot/Kotlin_Retrofit_Room_Data_Binding/Repository/PostInfoRoomDataBase.kt
package com.mvvm.kot.Kotlin_Retrofit_Room_Data_Binding.Repository

import android.arch.persistence.db.SupportSQLiteDatabase
import android.arch.persistence.room.Database
import android.arch.persistence.room.Room
import android.arch.persistence.room.RoomDatabase
import android.content.Context
import com.mvvm.kot.ViewModels.User

@Database(entities = arrayOf(ResultModel::class), version = 1,exportSchema = false)
abstract  class PostInfoRoomDataBase: RoomDatabase(){
    abstract fun postInfoDao(): PostInfoDao

    companion object {
        private var INSTANCE: PostInfoRoomDataBase? = null

        fun getInstance(context: Context): PostInfoRoomDataBase? {
            if (INSTANCE == null) {
                synchronized(PostInfoRoomDataBase::class) {
                    INSTANCE = Room.databaseBuilder(context,
                            PostInfoRoomDataBase::class.java, "post_info_database.db")
                            .addCallback(sRoomDataBaseCallback)
                            .build()
                }
            }
            return INSTANCE
        }

        fun destroyInstance() {
            INSTANCE = null
        }

        val sRoomDataBaseCallback = object:RoomDatabase.Callback(){
            override fun onOpen(db: SupportSQLiteDatabase) {
                super.onOpen(db)
                PopulateDbAsync(INSTANCE!!).execute()
            }
        };
     /* val sRoomDatabaseCallback = object : RoomDatabase.Callback() {

            override fun onOpen(db: SupportSQLiteDatabase) {
                super.onOpen(db)
                InitDBAsync(INSTANCE).execute()
            }
        }*/
    }

    }
